streams/src/main/java/org/apache/kafka/streams/kstream/EmitStrategy.java

```diff
@@ -19,15 +19,52 @@
 import org.apache.kafka.streams.kstream.internals.UnlimitedWindow;
 import org.apache.kafka.streams.kstream.internals.emitstrategy.WindowCloseStrategy;
 import org.apache.kafka.streams.kstream.internals.emitstrategy.WindowUpdateStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * This interface controls the strategy that can be used to control how we emit results in a processor.
  */
 public interface EmitStrategy {
 
+    Logger log = LoggerFactory.getLogger(EmitStrategy.class);
+
     enum StrategyType {
-        ON_WINDOW_CLOSE,
-        ON_WINDOW_UPDATE
+        ON_WINDOW_UPDATE(0, new WindowUpdateStrategy()),
+        ON_WINDOW_CLOSE(1, new WindowCloseStrategy());
+
+        private final short code;
+        private final EmitStrategy strategy;
+
+        private short code() {
+            return this.code;
+        }
+
+        private EmitStrategy strategy() {
+            return this.strategy;
+        }
+
+        StrategyType(final int code, final EmitStrategy strategy) {
```

Comment:
This is because Java's default type is int -- i.e. line 36/37 above would take the value 0/1 as int. So we basically need to either do the conversion in each line of 36/37 above, or just do the conversion once here.

I followed our other enums like `Errors` to do the conversion here.